### PR TITLE
Deploy site to Azure

### PR DESCRIPTION
**Pull request by pderop**
This is an attempt to deploy the site to Azure.

### CODE REVIEW

File: .github/workflows/deploy-to-azure.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    env:
+      SERVICE_NAME: ${{ secrets.AZURE_SERVICE_NAME }}
+      APP_NAME: ${{ secrets.AZURE_APP_NAME }}
+      JAR_PATH: ${{ github.workspace }}/build/libs/projectreactor-home.jar
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - uses: actions/setup-java@9704b39bf258b59bc04b50fa2dd55e9ed76b47a8 # v4
+        with:
+          distribution: 'temurin'
+          java-version: 8
```
Comment (chemicL) [inline .github/workflows/deploy-to-azure.yml:23]: Why 8?

File: .github/workflows/deploy-to-azure.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - uses: gradle/gradle-build-action@29c0906b64b8fc82467890bfb7a0a7ef34bda89e # v3
+        name: build
```
Comment (chemicL) [inline .github/workflows/deploy-to-azure.yml:25]: maybe the named steps should begin with name and then specify what action they use

File: .github/workflows/deploy-to-azure.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          arguments: build -x spotlessCheck
+      - name: Login via Azure CLI
+        uses: azure/login@v1
```
Comment (chemicL) [inline .github/workflows/deploy-to-azure.yml:29]: I suppose we should use a specific changeset

File: .github/workflows/deploy-to-azure.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS_SPRING_IO }}
+      - name: Create Staging Deploy
```
Comment (chemicL) [inline .github/workflows/deploy-to-azure.yml:32]: "Set to production" and "Delete staging environment" don't have all words capitalized, perhaps this step can be consistent

File: .github/workflows/deploy-to-azure.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          app-name:  ${{ env.APP_NAME }}
+          use-staging-deployment: true
+      - name: Delete staging deployment
+        uses: Azure/spring-apps-deploy@v1
+        with:
+          azure-subscription: ${{ secrets.AZURE_SUBSCRIPTION }}
+          action: delete-staging-deployment
+          service-name: ${{ env.SERVICE_NAME }}
+          app-name:  ${{ env.APP_NAME }}
+          use-staging-deployment: true
```
Comment (chemicL) [inline .github/workflows/deploy-to-azure.yml:58]: Is this argument needed?

File: .github/workflows/deploy-to-azure.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+          creds: ${{ secrets.AZURE_CREDENTIALS_SPRING_IO }}
+      - name: Create Staging Deploy
+        uses: Azure/spring-apps-deploy@v1
+        with:
+          azure-subscription: ${{ secrets.AZURE_SUBSCRIPTION }}
+          action: deploy
+          service-name: ${{ env.SERVICE_NAME }}
+          app-name:  ${{ env.APP_NAME }}
+          create-new-deployment: true
+          use-staging-deployment: true
```
Comment (chemicL) [inline .github/workflows/deploy-to-azure.yml:40]: Is this argument needed?

File: .github/workflows/deploy-to-azure.yml (hunk 3 above)
Comment (pderop) [reply .github/workflows/deploy-to-azure.yml:29]: I was actually expecting dependabot to suggest the correct SHA hash ?

File: .github/workflows/deploy-to-azure.yml (hunk 5 above)
Comment (pderop) [reply .github/workflows/deploy-to-azure.yml:58]: it's likely that we need this, see [this example](https://github.com/Azure/spring-apps-deploy/blob/main/.github/workflows/main.yml#L114) which I used.
(this PR is a first attempt, I need to test it and verify it's correctness)

File: .github/workflows/deploy-to-azure.yml (hunk 6 above)
Comment (pderop) [reply .github/workflows/deploy-to-azure.yml:40]: most likely, see [this example](https://github.com/Azure/spring-apps-deploy/blob/main/.github/workflows/main.yml#L82) I used.
but again, I could not test this workflow yet, so I need to make a first run and see if the site is correctly deployed.
maybe I'll have to redo more adaptations (crossing the fingers).
PS: the projectreactor.io domain is still pointing to our current active CF site ; so no worries ... it's just a first test.

File: .github/workflows/deploy-to-azure.yml (hunk 2 above)
Comment (pderop) [reply .github/workflows/deploy-to-azure.yml]: I have corrected this, see https://github.com/reactor/projectreactor.io/pull/147/commits/a427a6d46d85ee21b003df77a9efb7207e20b1f5

File: .github/workflows/deploy-to-azure.yml (hunk 4 above)
Comment (pderop) [reply .github/workflows/deploy-to-azure.yml:32]: I have corrected this, see https://github.com/reactor/projectreactor.io/commit/a427a6d46d85ee21b003df77a9efb7207e20b1f5

**Review by chemicL — APPROVED** [review]

**Pull request merged**